R/utils-json.R

```diff
@@ -165,7 +165,16 @@ getJSON <- function(.pd, evilMode) {
     namedAttrList$computedVariableMetadata <- computedVariableMetadata
   }
   
+  # Fix single data points in a single group for scatterplot and line plot - should have seriesX AND seriesY columns
+  # Fix only needed when we hae excatly one data point in our one group
+  if (nrow(.pd) == 1 && 'seriesX' %in% names(.pd) && length(.pd$seriesX) == 1) {
+    # Either one (see issue 144) or both columns may be single strings not lists. 
+    if (typeof(.pd$seriesX) != 'list') .pd$seriesX <- list(list(.pd$seriesX))
+    if (typeof(.pd$seriesY) != 'list') .pd$seriesY <- list(list(.pd$seriesY))
+  }
```

Comment:
are we VERY confident this only impacts scatter and line? not only is this a highly specific solution, but it references seriesX and seriesY directly outside of the classes to which they conceptually belong.

Reply:
yeah this solution was sort of my best of all the bad solutions i could think of. From the R perspective, the data in the .pd object looks the same whether it's this specific case of a single data point or a single data point in one of many groups. To me this meant I shouldn't put the fix within the plot class, because then i'd get out data that looks different in these different cases (one where we have a string wrapped in a list and one just a string). Which left making a fix in getJSON that annoyingly has to know about these specific columns.

Perhaps a more generalized solution would be that getJSON is passed info about columns and the api it should match? So then it would just check to see if the columns exist and that they are lists/characters/numbers/etc if they should be.

None of the above was an answer to your question :) which i'll go look into

Reply:
Finally answer to your question:
Tested all with one point and did not see the same issue in any other plot. Mosaic actually erred because of the chi-squared test. Otherwise all gave good responses.
Tested histo also with many data points but giant bin width to replicate how Bob arrived at his line plot error. All looked good.

This actually makes things odder... i did not expect those results. I'll keep thinking on this

I also thought about ensuring every non-strata non-error column in the data part of the output is a list if it's just one value, but r2 needs to be a string not an array.

Reply:
three thoughts:

1. do we know what it is making scatter and line / seriesX/Y special for this case?

Reply:
Oof finally found it! Explains both scatter and line. 
`lapply(.pd$seriesY, as.character)`
only happens for seriesY in line, and happens for both seriesX and Y in scatter. 
I'll fix it here!